tests/bluetooth/bsim_bt/bsim_test_iso/src/main.c

```diff
@@ -123,21 +124,217 @@ static void test_iso_main(void)
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+
+	printk("PER_ADV_SYNC[%u]: [DEVICE]: %s sync terminated\n",
+	       bt_le_per_adv_sync_get_index(sync), le_addr);
+
+	FAIL("PA terminated unexpectedly\n");
+}
+
+static void pa_recv_cb(struct bt_le_per_adv_sync *sync,
+		       const struct bt_le_per_adv_sync_recv_info *info,
+		       struct net_buf_simple *buf)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+
+	printk("PER_ADV_SYNC[%u]: [DEVICE]: %s, tx_power %i, "
+	       "RSSI %i, CTE %u, data length %u\n",
+	       bt_le_per_adv_sync_get_index(sync), le_addr, info->tx_power,
+	       info->rssi, info->cte_type, buf->len);
+}
+
+static struct bt_le_per_adv_sync_cb sync_cb = {
+	.synced = pa_sync_cb,
+	.term = pa_terminated_cb,
+	.recv = pa_recv_cb
+};
+
+static const char *phy2str(uint8_t phy)
+{
+	switch (phy) {
+	case 0: return "No packets";
+	case BT_GAP_LE_PHY_1M: return "LE 1M";
+	case BT_GAP_LE_PHY_2M: return "LE 2M";
+	case BT_GAP_LE_PHY_CODED: return "LE Coded";
+	default: return "Unknown";
+	}
+}
+
+static void scan_recv(const struct bt_le_scan_recv_info *info,
+		      struct net_buf_simple *buf)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+	char name[30];
+
+	/* We only care for scan results until we find the SID */
+	if (per_sid) {
+		return;
+	}
+
+	(void)memset(name, 0, sizeof(name));
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	printk("[DEVICE]: %s, AD evt type %u, Tx Pwr: %i, RSSI %i %s "
+	       "C:%u S:%u D:%u SR:%u E:%u Prim: %s, Secn: %s, "
+	       "Interval: 0x%04x (%u ms), SID: %u\n",
+	       le_addr, info->adv_type, info->tx_power, info->rssi, name,
+	       (info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) != 0,
+	       (info->adv_props & BT_GAP_ADV_PROP_SCANNABLE) != 0,
+	       (info->adv_props & BT_GAP_ADV_PROP_DIRECTED) != 0,
+	       (info->adv_props & BT_GAP_ADV_PROP_SCAN_RESPONSE) != 0,
+	       (info->adv_props & BT_GAP_ADV_PROP_EXT_ADV) != 0,
+	       phy2str(info->primary_phy), phy2str(info->secondary_phy),
+	       info->interval, info->interval * 5 / 4, info->sid);
+
+	if (info->interval) {
+		per_sid = info->sid;
+		bt_addr_le_copy(&per_addr, info->addr);
+	}
+}
+
+static struct bt_le_scan_cb scan_callbacks = {
+	.recv = scan_recv,
+};
+
+static void test_iso_recv_main(void)
+{
+	int err;
+
+	printk("\n*ISO broadcast test*\n");
+
+	printk("Bluetooth initializing...\n");
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Could not init BT: %d\n", err);
+		return;
+	}
+	printk("success.\n");
+
+	printk("Scan callbacks register...");
+	bt_le_scan_cb_register(&scan_callbacks);
+	printk("success.\n");
+
+	printk("Periodic Advertising callbacks register...");
+	bt_le_per_adv_sync_cb_register(&sync_cb);
+	printk("Success.\n");
+
+	printk("Start scanning...");
+	err = bt_le_scan_start(BT_LE_SCAN_ACTIVE, NULL);
+	if (err) {
+		FAIL("Could not start scan: %d\n", err);
+		return;
+	}
+	printk("success.\n");
+
+#if 0
```

Comment:
The test case seems to stop here after starting to scan. Shouldn't it go somewhat further to exercise the new functionality?

Reply:
At this point there isn't much more to exercise. With no LLL support/integration yet, we cannot test if there's anything incoming. 

We could start at periodic advertising on the broadcaster, and a scan on the receiver and even create a periodic advertising sync (which will be needed in order to get the BIGInfo data) and then terminate that, but I'd rather expand on that in a separate PR. This test has mainly been added to ensure that the functions and tickers can be correctly initialized.